testsuite/testsuite.py

```diff
@@ -62,8 +66,16 @@ def tear_up(self):
         self.env.repo_base = os.path.abspath(os.path.join(
             os.path.dirname(__file__), '..'))
 
+        self.env.wait_factor = 1
+
         # Absolute paths to programs that test drivers can use
-        self.env.als = self.lookup_program('server', 'ada_language_server')
+        if self.env.options.valgrind_memcheck:
+            self.env.als = "{} --tool=memcheck --quiet {}".format(
```

Comment:
Not sure why you want it to be quiet? Having the diff with the locations of the leaks (even passing `--leak-check=full`) would help in nightly runs/CIs IMHO.

Reply:
Valgrind’s `--quiet` option is used to remove all the messages that are not memory issues diagnostics: leaks, invalid reads and so on (with their stack traces) are still printed.

Reply:
@pmderodat aah ok! Still, in this setup you won't get all the diagnostics that you get from `--leak-check=full` right?

Reply:
You’re right: `--leak-check=full` would help to find more memory leaks.

Reply:
ah, note that this mode "memcheck" is going after memory corruption, not leaks! I want to only display a message if there's an error. I will add a mode valgrind_leaks for the leaks, in a separate PR.

Reply:
Why doing two different modes?  `leak-check=full` is the same checker (memcheck) and will also detect memory corruptions